opencensus/metrics/export/gauge.py

```diff
@@ -16,14 +16,48 @@
 import six
 import threading
 
+from opencensus.common import utils
 from opencensus.metrics.export import metric
 from opencensus.metrics.export import metric_descriptor
 from opencensus.metrics.export import point as point_module
 from opencensus.metrics.export import time_series
 from opencensus.metrics.export import value as value_module
 
 
-class GaugePointLong(object):
+def get_timeseries_list(points, timestamp):
+    """Convert a list of `GaugePoint`s into a list of `TimeSeries`.
+
+    Get a :class:`opencensus.metrics.export.time_series.TimeSeries` for each
+    measurement in `points`. Each series contains a single
+    :class:`opencensus.metrics.export.point.Point` that represents the last
+    recorded value of the measurement.
+
+    :type points: list(:class:`GaugePoint`)
+    :param points: The list of measurements to convert.
+
+    :type timestamp: :class:`datetime.datetime`
+    :param timestamp: Recording time to report, usually the current time.
+
+    :rtype: list(:class:`opencensus.metrics.export.time_series.TimeSeries`)
+    :return: A list of one `TimeSeries` for each point in `points`.
+    """
+    ts_list = []
+    for lv, gp in points.items():
+        point = point_module.Point(gp.to_point_value(), timestamp)
+        ts_list.append(time_series.TimeSeries(lv, [point], timestamp))
+    return ts_list
+
+
+class GaugePoint(object):
```

Comment:
We could probably put the __repr__ implementation in the base class.

Reply:
Probably extract the lock initialization in to `GaugePoint.__init__` as well?

```
class GaugePoint(object):
    def __init__(self):
        self._value_lock = threading.Lock()
```

Reply:
I didn't do this because I wanted `DerivedGaugePoint` to be a `GaugePoint` too, and this would mean that we couldn't call its parent's constructor (or create an unused lock if we did).

I think there's good reason to have abstract classes like these where all the methods are unimplemented, but you can argue that classes should either be abstract or concrete -- and not like `Gauge` and `BaseGauge` that implement some methods and not others. What do you think?

Reply:
I see, makes sense.

I don't have strong opinion here, both approaches have their valid reasons and I don't see a clear winner. Go with your choice :)